### PR TITLE
vehicles: Update doc/JSON_INFO.md for all fields in vehicle_part

### PR DESCRIPTION
**Pull request by mlangsdorf**
#### Summary
```SUMMARY: Build "vehicles: Update doc/JSON_INFO.md for all fields in vehicle_part"```

#### Purpose of change
The JSON_INFO.md for vehicle parts has gotten out of date, so update it to the current code base and include all fields.

#### Describe the solution
Documentation only change.

### CODE REVIEW

File: doc/JSON_INFO.md [hunk 1]
```diff
@@ -1815,52 +1815,136 @@ Vehicle components when installed on a vehicle.
...
+                              // Only the "LIFT", "JACK", and "SELF_JACK" qualities are valid.
+"transform_terrain": {        // (Optional) This part can transform terrain, like a plow.
+                              // One of "post_terain", "post_furniture", or "post_field" is required.
+  "pre_flags": [ "PLOWABLE" ], // List of flags for the terrain that can be transformed.
+  "post_terrain": "t_dirtmound", // (Optional, default to "t_null") The resulting terrain, if any.
+  "post_furniture": "f_boulder", // (Optional, default to "f_null") The resulting furniture, if any.
+  "post_field": "fd_fire",    // (Optional, default to "fd_null") The resulting field, if any.
+  "post_field_intensity": 10, // (Optional, default to 0) The field's intensity, if any.
+  "post_field_age": "20 s"    // (Optional, default to 0 turns) The field's time to live, if any.
+o},
```
Comment (wapcaplet) [inline doc/JSON_INFO.md:1873]:
```suggestion
},
```

File: doc/JSON_INFO.md [hunk 2]
```diff
@@ -1815,52 +1815,136 @@ Vehicle components when installed on a vehicle.
...
-"exclusions": [ "souls" ]     // Optional field, defaults to empty. A list of words. A new engine can't be installed on the vehicle if any engine on the vehicle shares a word from exclusions.
-"fuel_options": [ "soul", "black_soul" ] // Optional field, defaults to fuel_type.  A list of words. An engine can be fueled by any fuel type in its fuel_options.  If provided, it overrides fuel_type and should include the fuel in fuel_type.
-"comfort": 3,                 // Optional field, defaults to 0. How comfortable this terrain/furniture is. Impact ability to fall asleep on it. (uncomfortable = -999, neutral = 0, slightly_comfortable = 3, comfortable = 5, very_comfortable = 10)
-"floor_bedding_warmth": 300,  // Optional field, defaults to 0. Bonus warmth offered by this terrain/furniture when used to sleep.
-"bonus_fire_warmth_feet": 200,// Optional field, defaults to 300. Increase warmth received on feet from nearby fire.
+ "qualities": [ [ "SELF_JACK", 17 ] ], // (Optional) A list of lists, with each list being a tool
```
Comment (wapcaplet) [inline doc/JSON_INFO.md:1862]:
```suggestion
"qualities": [ [ "SELF_JACK", 17 ] ], // (Optional) A list of lists, with each list being a tool
```

File: doc/JSON_INFO.md (hunk 1 above)
Comment (mlangsdorf) [reply doc/JSON_INFO.md]: Thanks.

File: doc/JSON_INFO.md (hunk 2 above)
Comment (mlangsdorf) [reply doc/JSON_INFO.md]: Thankis.

**Pull request merged**